.gitlab-ci.yml

```diff
@@ -533,6 +533,60 @@ dockerize:dockerExtensions:
     - docker login -u gitlab-ci-token -p $CI_JOB_TOKEN $CI_REGISTRY
     - yarn run in-submodules -- -f categories.dockerExtension=true -- run docker-build-prod --include-filtered-dependencies -- -- --repository=$CI_REGISTRY/magda-data/magda --version=$CI_COMMIT_REF_SLUG --cacheFromVersion=master
 
+
+
+inttest:registryAuth:
+  # https://github.com/kind-ci/examples/blob/master/.gitlab-ci.yml
+  stage: preview
+  retry: 1
+  image: registry.gitlab.com/magda-data/magda/data61/magda-builder-scala:$CI_COMMIT_REF_SLUG
+  allow_failure: true
+  dependencies:
+    - sbt-prebuild
+    - builders-and-yarn
+    - buildtest:registry
+  services:
+    - docker:dind
+  cache:
+    key: $CI_JOB_NAME-$CACHE_VERSION
+    paths:
+      - /usr/local/bin/kind
+      - /usr/local/bin/kubectl
+      - /usr/local/bin/helm
+  variables:
+    KUBECTL: v1.15.0
+    KIND: v0.7.0
+  script:
+    - apk add --update postgresql-client sed curl
+    - curl -Lo /usr/local/bin/kind https://github.com/kubernetes-sigs/kind/releases/download/${KIND}/kind-$(uname)-amd64 -C -
+    - chmod +x /usr/local/bin/kind
+    - curl -Lo /usr/local/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/${KUBECTL}/bin/linux/amd64/kubectl -C -
+    - chmod +x /usr/local/bin/kubectl
+    - curl -L https://storage.googleapis.com/kubernetes-helm/helm-v2.16.1-linux-amd64.tar.gz -C - | tar -xzO linux-amd64/helm > /usr/local/bin/helm
+    - chmod +x /usr/local/bin/helm
+    - kind create cluster --config=./kind-config.yaml --wait 20m --image kindest/node:v1.15.7@sha256:e2df133f80ef633c53c0200114fce2ed5e1f6947477dbc83261a6a921169488d
+    # Set up KUBECONFIG that links to kind - we use sed to edit it so it's pointing to gitlab's docker host
+    - kind get kubeconfig | sed -E -e 's/localhost|0\.0\.0\.0/docker/g' > ./kubeconfig
+    - export KUBECONFIG="$(pwd)/kubeconfig"
+    - kubectl apply -f deploy/kubernetes/rbac-config.yaml
+    - helm init --service-account tiller --wait
+    - export JWT_SECRET=udIsbcYaKs1G4n6AdiMSIvPx5KpxQAy8FA2aIcD46iCipNAZvds4jeXFLZKhVvSJZvhYb5Pvgvmtonk7UFfhGnYcd3DXM7KzHG7gBmGO8PCsOZ4t7icqZoJbpdDqYWMmd9XnrVXtJhR6HVFBmEmbk9AmFJ1Gz9ipYPGYLoFcavPs9iZ63KPXgdt4aBdWQcmICkGPYiY8CQOvqOoiU7hUhKDTkJgRRTSaax6UQDOveTQvQnd5uyXuV4os0tlahzRX
+    - kubectl create ns test2
+    - 'echo "{ \"apiVersion\": \"v1\", \"kind\": \"Secret\", \"metadata\": {\"name\": \"auth-secrets\"}, \"type\": \"Opaque\", \"data\": {\"jwt-secret\": \"dWRJc2JjWWFLczFHNG42QWRpTVNJdlB4NUtweFFBeThGQTJhSWNENDZpQ2lwTkFadmRzNGplWEZMWktoVnZTSlp2aFliNVB2Z3ZtdG9uazdVRmZoR25ZY2QzRFhNN0t6SEc3Z0JtR084UENzT1o0dDdpY3Fab0picGREcVlXTW1kOVhuclZYdEpoUjZIVkZCbUVtYms5QW1GSjFHejlpcFlQR1lMb0ZjYXZQczlpWjYzS1BYZ2R0NGFCZFdRY21JQ2tHUFlpWThDUU92cU9vaVU3aFVoS0RUa0pnUlJUU2FheDZVUURPdmVUUXZRbmQ1dXlYdVY0b3MwdGxhaHpSWA==\", \"session-secret\":\"squirrel\"}}" | kubectl apply --namespace test2 -f -'
```

Comment:
Not sure whether it will eventualy create any trouble.
But as it's `Opaque` type, "session-secret\":\"squirrel\" 's value should probably be base64 encoded as `c3F1aXJyZWw=` or create secret as:
```
kubectl create secret generic auth-secrets --from-literal=jwt-secret=$JWT_SECRET --from-literal=session-secret=squirrel --namespace test2 --dry-run -o json | kubectl apply --namespace test2 -f -
```

Reply:
Yeah that makes sense - the tests don't actually need `session-secret` at all (it's a gateway thing) so I've just removed it :)